### ISSUE TITLE
Handling Unbalanced Classes in GPFlow

### ISSUE THREAD

**Issue by hoangcuong2011** [GPflow/GPflow#618]
Hi all,

I am about running experiments with Gaussian Processes for classification, using the GPFlow engine.

I got stuck with handling unbalanced classes. I know there are several different approaches to deal with the problem, and I am trying to modify the loss function with suitable associated weights to do so.

However, I could not find any online documents/references to move on. I noticed for some (unrelated) engines such as Keras, it is relative easy to do so https://datascience.stackexchange.com/questions/13490/how-to-set-class-weights-for-imbalanced-classes-in-keras. But those are for neural networks. Is there such a way in GPflow for training GPs, and if it is, could you give me some hints?

Thanks a lot,

Best,

Hoang Cuong.

**Comment by markvdw** [comment]
How do you want to deal with unbalanced classes? If the classes are unbalanced in the test set, you don't need to take any further action than training.

We currently don't have any built in mechanism like the one you mention. You can hack together something by simply repeating data points a certain number of times. Alternatively, you could modify the objective functions themselves to weight points of certain labels more.
